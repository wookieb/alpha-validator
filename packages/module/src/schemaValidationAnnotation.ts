import * as is from 'predicates';

/**
 * Annotation that marks given service as validation function for given schema names
 */
export function schemaValidationAnnotation(...schemaNames: string[]): Annotation {
    is.assert(is.notEmptyArray, 'You need to provide at least one schema name')(schemaNames);
    is.assert(is.arrayOf(is.notBlank), 'Every schema name must be a non-blank string')(schemaNames);
    return {
        name: NAME,
        schemaNames
    };
}

const NAME = 'alpha-validator/schema-validation';

export interface Annotation {
    name: string;
    schemaNames: string[];
}

export const PREDICATE = is.prop('name', is.eq(NAME));