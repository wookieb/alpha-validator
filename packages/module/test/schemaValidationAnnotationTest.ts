import {schemaValidationAnnotation} from "@src/schemaValidationAnnotation";

describe('schemaValidationAnnotation', () => {
    describe('failure', () => {
        function assert(schemaNames: any) {
            expect(() => {
                schemaValidationAnnotation(schemaNames);
            })
                .toThrowErrorMatchingSnapshot();
        }
        describe('requires schema name', () => {
            it('to be a non-blank string', () => {
                assert('   ');
            });

            it('to be a non-empty string', () => {
                assert('');
            });

            it('to be a string', () => {
                assert(true);
            });
        });
    });

    describe('success', () => {
        it('registering single schema', () => {
            expect(schemaValidationAnnotation('schema1'))
                .toMatchSnapshot();
        });

        it('registering multiple schemas', () => {
            expect(schemaValidationAnnotation('schema1', 'schema2'))
                .toMatchSnapshot();
        });
    });
});