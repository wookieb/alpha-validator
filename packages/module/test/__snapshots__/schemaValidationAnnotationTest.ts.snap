// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schemaValidationAnnotation failure requires schema name to be a non-blank string 1`] = `"Every schema name must be a non-blank string"`;

exports[`schemaValidationAnnotation failure requires schema name to be a non-empty string 1`] = `"Every schema name must be a non-blank string"`;

exports[`schemaValidationAnnotation failure requires schema name to be a string 1`] = `"Every schema name must be a non-blank string"`;

exports[`schemaValidationAnnotation success registering multiple schemas 1`] = `
Object {
  "name": "alpha-validator/schema-validation",
  "schemaNames": Array [
    "schema1",
    "schema2",
  ],
}
`;

exports[`schemaValidationAnnotation success registering single schema 1`] = `
Object {
  "name": "alpha-validator/schema-validation",
  "schemaNames": Array [
    "schema1",
  ],
}
`;
